jdaviz/app.py

```diff
@@ -1330,7 +1339,9 @@ def _on_data_added(self, msg):
             The Glue data collection add message containing information about
             the new data.
         """
-        self._link_new_data()
+        # We don't need to link the first data to itself
+        if len(self.data_collection) > 1:
+            self._link_new_data()
```

Comment:
just trying to wrap my head around the need for all the changes in tests - is avoiding self-linking (alone) responsible for dropping all those links and affecting the indices?

Reply:
I believe so, yes. In Cubeviz there were 3 extra links, Specviz 1 extra. Hence the changes.